### PR TITLE
Strip TypeLoc out of Patterns

### PR DESCRIPTION
**Pull request by CodaFi**
Strip it out of

* IsPattern
* EnumElementPattern
* Reduce usage of TypedPattern::getTypeLoc

### CODE REVIEW

**Comment by CodaFi** [comment]
@swift-ci clean smoke test macOS

File: lib/Sema/TypeCheckPattern.cpp [hunk 1]
```diff
@@ -671,44 +671,46 @@ Pattern *TypeChecker::resolvePattern(Pattern *P, DeclContext *DC,
...
-    hadError = TypeChecker::validateType(TL, resolution);
+
+    auto *var = named->getDecl();
+    auto opaqueDecl = var->getOpaqueResultTypeDecl();
+    if (!opaqueDecl) {
+      return ErrorType::get(Context);
+    }
+
+    auto opaqueTy = opaqueDecl->getDeclaredInterfaceType();
+    if (!opaqueTy || opaqueTy->hasError()) {
```
Comment (AnthonyLatsis) [inline lib/Sema/TypeCheckPattern.cpp:700]: Can this ever be a null type?

**Comment by CodaFi** [comment]
@swift-ci smoke test

**Comment by CodaFi** [comment]
@swift-ci smoke test

**Comment by CodaFi** [comment]
⛵

**Pull request merged**